docs/modules/ROOT/pages/concepts/meter-provider.adoc

```diff
@@ -0,0 +1,49 @@
+[[meter-provider]]
+= Meter Provider
+
+It's a common use-case to attach tags dynamically to a `Meter`. Let's say we execute a job and we want to use a `Timer` to instrument it:
+
+[source, java]
+----
+Timer.Sample sample = Timer.start(registry);
+
+Result result = job.execute();
+
+Timer timer = Timer.builder("job.execution")
+    .tag("job.name", "job")
+    .tag("status", result.status())
+    .register(registry);
+sample.stop(timer);
+----
+
+This lets us dynamically determine the `status` tag from the end state of the operation we are timing. There are two drawbacks of doing this:
+
+1. Every time the above is executed, a new `Timer.Builder` instance is created. This increases the amount of data that the GC needs to collect.
+2. The code above is somewhat boilerplate, it does not let you define the common properties of a Timer and attach what is dynamically changing but everything is always present.
+
+NOTE: In some cases you can use `registry.timer("job.execution", "job.name", "my-job", "status", result.status())` instead of using `Timer.Builder` which can save you some extra objects but this is not always possible.
+
+You can resolve both of these issues by using a `MeterProvider`. It's a convenience interface to create new meters from tags using a common "template"/"builder".
```

Comment:
I would avoid "builder" to make it as distinct as possible from the actual Timer.Builder et al.

Reply:
👍🏼
```suggestion
You can resolve both of these issues by using a `MeterProvider`. It's a convenience interface to create new meters from tags using a common "template".
```